tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+    &lt;div @selector="content" xmlns="http://www.w3.org/1999/xhtml"&gt;
+        &amp;copy; // copy HTML entity name will throw an error ❌
+    &lt;/div&gt;
+&lt;/foreignObject&gt;
+
+...
+
+// Use an entity number ✅
+&amp;#169;
+</code></pre>
+        
+        <h4>Browser related quirks</h4>
+
+        <p>
+            At the beginning of this tutorial, we mentioned how <code>foreignObject</code> is fully supported in all browsers.
+            Unfortunately, that doesn't mean some inconsistencies exist, or that some unexpected behaviour might occur. 
+        </p>
+
+        <p>
+            Our experience with both Chrome and Firefox has been very positive, but we have ran into some quirks when using Safari. 
+            In this section, we will cover some issues we discovered, and some related 
+            <a href="https://bugs.webkit.org/query.cgi" target="_blank">webkit</a> bugs, but this also isn't an exhaustive list, so you 
+            might run into an issue with Safari that isn't mentioned here.
+        </p>
+
+        <p>
+            The first issue arose when we tried to add an HTML5 <code>video</code> element to a shape. While this went as expected
+            with Chrome and Firefox, the video seemed to be rendered in an incorrect position on Safari. After searching for a solution,
+            we stumbled on the following <a href="https://bugs.webkit.org/show_bug.cgi?id=23113" target="_blank">webkit bug</a>.
+        </p>
+
+        <p>
+            Upon reading, we couldn't help but notice others suggesting that properties like <code>opacity</code>, or <code>transform</code>
+            were also causing issues for Safari users, so we added those styles to our elements, and sure enough, we saw similar results.
+            HTML elements which had <code>opacity</code> or <code>transform</code> applied were also in the incorrect position.
+        </p>
+
+        <p>
+            Naturally, we also searched the same bug report for any potential solutions, and found suggestions that adding a style of
+            <code>'position: fixed'</code> to the outermost <code>div</code> element fixed some of the issues that people had. Once again,
+            the information proved to be helpful, and the elements seemed to be positioned correctly again.
+        </p>
+
+        <p>
+            On further inspection, we realized that we still had one major issue with the video. After dragging our JointJS shape, the video
+            was still rendered in the incorrect position which was disappointing. The silver lining was that the <code>opacity</code> and 
+            <code>transform</code> applied to our <code>label</code> and <code>input</code> respectively were working correctly, and those 
+            elements seemed to maintain the correct position. The following is some example <code>markup</code> we had issues with in Safari, 
+            your mileage may vary.
```

Comment:
i think these 4 paragraphs are confusing to the reader - i don't think they care about what we did, just about what the problem is and what the solution is... can you rework this section?

i would suggest replacing these paragraphs with a list where each item has: problem statement - webkit bug number - workaround - issues with workaround

i think you are describing three issues, right (see, i'm not even sure about that) - video issue 1, opacity+transform issue, and video issue 2... having them as a three-item list would be much clearer

Reply:
I will keep this `browser related quirks` section as it is. A lot of how the section is structured is to show we encountered issues mostly through trial and error, protect us from questions as much as possible, and allow for maximum flexibility in updating it.

These paragraphs were written in this way intentionally with the hope of encouraging people to find solutions themselves rather than coming to us with questions. I want to be as non-committal as possible with issues, workarounds, etc. If we have to add more elements or CSS properties, the current structure allows me to update the section in the most flexible way.